Download Python test currently failing

```
$ pytest -k test_download_python
============================= test session starts ==============================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
rootdir: /home/ethan/Dropbox/Informatics/offlinedatasci
plugins: mock-3.12.0
collected 4 items / 3 deselected / 1 selected                                  

test/test_offlinedatasci.py F                                            [100%]

=================================== FAILURES ===================================
_____________________________ test_download_python _____________________________

tmp_path = PosixPath('/tmp/pytest-of-ethan/pytest-0/test_download_python0')

    def test_download_python(tmp_path):
>       download_python(tmp_path)

test/test_offlinedatasci.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

ods_dir = PosixPath('/tmp/pytest-of-ethan/pytest-0/test_download_python0')

    def download_python(ods_dir):
        """Download Python installers from HTML page
    
        Keyword arguments:
        ods_dir -- Directory to save installers
        """
        url = get_python_download_page()
        download_table_num=0
        oscolnum=1
        hrefcolnum=0
        key="version"
    
        destination_path = Path(Path(ods_dir), Path("python"))
        if not os.path.isdir(destination_path):
            os.makedirs(destination_path)
        python_versions = {}
        fp = urllib.request.urlopen(url)
        web_content = fp.read()
        soup = bs.BeautifulSoup(web_content, 'lxml')
>       r_studio_download_table = soup.find_all('table')[download_table_num]
E       IndexError: list index out of range

../../../miniconda3/envs/ods/lib/python3.11/site-packages/offlinedatasci/main.py:220: IndexError
=========================== short test summary info ============================
FAILED test/test_offlinedatasci.py::test_download_python - IndexError: list index out of range
======================= 1 failed, 3 deselected in 0.58s ========================
```

--- a/offlinedatasci/main.py
+++ b/offlinedatasci/main.py
@@ -11,6 +11,7 @@ import subprocess
 import urllib.request, urllib.error, urllib.parse
 import importlib_resources
 import pypi_mirror
+import requests
 import shutil
 import sys
 import warnings
@@ -189,8 +190,8 @@ def download_rstudio(ods_dir):
     destination_path = Path(Path(ods_dir), Path("rstudio"))
     if not os.path.isdir(destination_path):
         os.makedirs(destination_path)
-    fp = urllib.request.urlopen(baseurl)
-    web_content = fp.read()
+    fp = requests.get(baseurl)
+    web_content = fp.content
     soup = bs.BeautifulSoup(web_content, 'lxml')
     links = soup.find_all('a')
     for link in links:
@@ -214,8 +215,8 @@ def download_python(ods_dir):
     if not os.path.isdir(destination_path):
         os.makedirs(destination_path)
     python_versions = {}
-    fp = urllib.request.urlopen(url)
-    web_content = fp.read()
+    fp = requests.get(url)
+    web_content = fp.content
     soup = bs.BeautifulSoup(web_content, 'lxml')
     r_studio_download_table = soup.find_all('table')[download_table_num]
     table_body = r_studio_download_table.find('tbody')
@@ -239,7 +240,7 @@ def find_r_current_version(url):
     url -- CRAN r-project URL
     """
     version_regex = "(R\-\d+\.\d+\.\d)+\-(?:x86_64|arm64|win)\.(?:exe|pkg)"
-    urlfile = urllib.request.urlopen(url)
+    urlfile = requests.get(url)
     for line in urlfile:
         decoded = line.decode("utf-8") 
         match = re.findall(version_regex, decoded)
@@ -297,8 +298,8 @@ def get_ods_dir(directory=Path.home()):
 def get_python_download_page():
     """Get download page from Python homepage."""
     base_url="https://www.python.org"
-    fp = urllib.request.urlopen(base_url)
-    web_content = fp.read()
+    fp = requests.get(base_url)
+    web_content = fp.content
     soup = bs.BeautifulSoup(web_content, "html.parser")
     release_a_tag = soup.find("a", href=lambda href: href and "release" in href)
     current_release_path = release_a_tag["href"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -43,6 +43,7 @@ dependencies = [
   'importlib_resources',
   'lxml',
   'python-pypi-mirror',
+  'requests',
   'setuptools'
 ]
 

